Apply ellipsis for overflowing content #31392

VANSH3104 opened the pull request:
<!-- Describe your pull request here.-->
#31392 
Fixes: <!-- Issue link, or clear description.-->
This PR resolves the issue of long text in dropdown options by converting overflowing text into ellipsis (...). It ensures that content such as long user names, topic names, and other dynamic text is neatly truncated when it exceeds the available space, improving the user interface's readability and appearance.


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

before
![IMG_20250122_024905](https://github.com/user-attachments/assets/124e2191-92e0-49e4-8a47-2d3f969ae980)
![IMG_20250122_025147](https://github.com/user-attachments/assets/99f37028-cf13-4424-a553-ee01628d6b91)


after

![IMG_20250122_025058](https://github.com/user-attachments/assets/b9417021-30dd-4832-b135-f3f77ea5282a)
![IMG_20250122_024933](https://github.com/user-attachments/assets/d312fd26-9ed9-463e-8b5f-ff36276cdaef)

## More images
![Screenshot from 2025-01-22 16-47-58](https://github.com/user-attachments/assets/2f861398-9fe3-4c58-8295-e2b1363fec27)
![Screenshot from 2025-01-22 16-47-07](https://github.com/user-attachments/assets/b2344fa2-5379-4c75-a07f-478e2d588beb)
![Screenshot from 2025-01-22 16-46-15](https://github.com/user-attachments/assets/3681c7aa-1246-4160-b425-d1a7c220a360)
![Screenshot from 2025-01-22 17-12-22](https://github.com/user-attachments/assets/d08cd3d3-cd58-4fe3-b34c-4b4fbfade417)
![Screenshot from 2025-01-22 17-09-59](https://github.com/user-attachments/assets/4a5f7a90-ddc0-4979-b5c5-11c677e172b0)

### The sentence should not have ellipses everywhere; correction is needed only where it is required."

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

VANSH3104 commented:
@alya i solved this issue but not able to understanding why cli fails I am not able to understand the logic of failure also .tools/lint also not giving any error

reharsh commented:
hii @VANSH3104, looks like your node tests are failing with 
<details><summary>error</summary>

```
test failed: /__w/zulip/zulip/web/tests/composebox_typeahead.test.cjs > initialize

AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+ actual - expected

  '    <span class="zulip-icon zulip-icon-user-circle-offline user-circle-offline user-circle"></span>\n' +
    '    <img class="typeahead-image" src="/avatar/101" />\n' +
    '<div class="typeahead-text-container">\n' +
+   '    <strong class="typeahead-strong-section">Othello, the Moor of...</strong>    <span class="autocomplete_secondary">othello@zulip.com</span></div>\n'
-   '    <strong class="typeahead-strong-section">Othello, the Moor of Venice</strong>    <span class="autocomplete_secondary">othello@zulip.com</span></div>\n'

    at new_value.<anonymous> (/__w/zulip/zulip/web/tests/composebox_typeahead.test.cjs:60:669)
    at new new_value (/__w/zulip/zulip/web/tests/lib/namespace.cjs:59:173)
    at initialize_compose_typeahead (/__w/zulip/zulip/web/src/composebox_typeahead.ts:1[33](https://github.com/zulip/zulip/actions/runs/12856477631/job/35843195570?pr=33091#step:12:34)4:9)
    at Proxy.initialize (/__w/zulip/zulip/web/src/composebox_typeahead.ts:1446:5)
    at /__w/zulip/zulip/web/tests/composebox_typeahead.test.cjs:69:256
    at /__w/zulip/zulip/web/tests/composebox_typeahead.test.cjs:6:1429
    at /__w/zulip/zulip/web/tests/lib/test.cjs:2:1414
    at Object.exports.with_overrides (/__w/zulip/zulip/web/tests/lib/namespace.cjs:67:186)
    at execute_test (/__w/zulip/zulip/web/tests/lib/test.cjs:2:1330)
    at /__w/zulip/zulip/web/tests/lib/test.cjs:3:227
    at run_one_module (/__w/zulip/zulip/web/tests/lib/index.cjs:7:488)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at /__w/zulip/zulip/web/tests/lib/index.cjs:8:114 {
  generatedMessage: true,
  code: 'ERR_ASSERTION',
  actual: '    <span class="zulip-icon zulip-icon-user-circle-offline user-circle-offline user-circle"></span>\n' +
    '    <img class="typeahead-image" src="/avatar/101" />\n' +
    '<div class="typeahead-text-container">\n' +
    '    <strong class="typeahead-strong-section">Othello, the Moor of...</strong>    <span class="autocomplete_secondary">othello@zulip.com</span></div>\n',
  expected: '    <span class="zulip-icon zulip-icon-user-circle-offline user-circle-offline user-circle"></span>\n' +
    '    <img class="typeahead-image" src="/avatar/101" />\n' +
    '<div class="typeahead-text-container">\n' +
    '    <strong class="typeahead-strong-section">Othello, the Moor of Venice</strong>    <span class="autocomplete_secondary">othello@zulip.com</span></div>\n',
  operator: 'strictEqual'
}
```
</details> 

you have to modify `composebox_typeahead.test.cjs` to account for ellipsis(...) you have added to pass them.

also couple of pointers:
- please don't add pr no. #xxxxx in the title and make sure your commits follow [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#write-a-clean-commit-history)
- there are test suits other than `./tools/lint` you should run locally too, checkout [test docs](https://zulip.readthedocs.io/en/latest/testing/testing.html#running-tests).

VANSH3104 commented:
hi @alya I've resolved the issue. Could you please review it when you have a moment? Once it's good to go, I'll move on to the next task.

alya commented:
As requested in the PR template, please post screenshots of all your changes in the PR description. It is very difficult to review a video.

You will also need to go through the self-review checklist.

VANSH3104 commented:
@alya I updated my commit and add images of changes I also self review checklist too

alya commented:
Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

You will also need to add a screenshot of the topics typeahead.

VANSH3104 commented:
Hi @alya,I’ve cleaned up the commit history and also added the screenshot of the topics typeahead as requested. I’ve reviewed everything and ensured that the changes were applied only where necessary. Please let me know if any further adjustments are needed.

alya commented:
@amanagr Are you up for reviewing this one?

amanagr commented:
We want to use a mostly CSS approach here. We don't want a hard code JS approach

VANSH3104 commented:
@amanagr ok I will correct it

VANSH3104 commented:
@amanagr  i just having a git error and my pr get closed I will make a pr again with the proper corrections
